src/txn/txn_ckpt.c

```diff
@@ -449,6 +449,16 @@ __wt_checkpoint_get_handles(WT_SESSION_IMPL *session, const char *cfg[])
     if (F_ISSET(btree, WT_BTREE_NO_CHECKPOINT | WT_BTREE_IN_MEMORY) || WT_IS_HS(btree->dhandle))
         return (0);
 
+    if (__wt_conn_is_disagg(session)) {
+        /* Skip the shared metadata table for disaggregated storage; we'll checkpoint it later. */
+        if (WT_IS_DISAGG_META(btree->dhandle))
+            return (0);
+        /* Skip checkpointing shared tables if we are not a leader. */
+        /* TODO: Figure out how to avoid checkpointing clean objects in *all* cases. */
```

Comment:
What does this todo mean? Are we not skipping clean objects?

Reply:
There are certain conditions under which WT includes a clean table in the global checkpoint, e.g., if it decides that by doing so it can free up some space (e.g., due to how the extent lists work). I originally thought that I was running into this condition, but I don't think that's actually the case anymore—I should remove this TODO.

Reply:
Fixed in [595f67f](https://github.com/wiredtiger/wiredtiger/pull/11211/commits/595f67f13c1042cd5b286e63252c89647b9a6e07).